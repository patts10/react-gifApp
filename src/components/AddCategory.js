import React, { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ setCategories }) => {

  const [inputvalue, setInputvalue] = useState("");

  const handleInputChange = (e) => {
    setInputvalue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();

    if (inputvalue.trim().length > 2) {
      setCategories((c) => [inputvalue, ...c]);
      setInputvalue("");
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <input
        className="my-5"
        type="text"
        value={inputvalue}
        onChange={handleInputChange}
        placeholder={"Search..."}
      />
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
