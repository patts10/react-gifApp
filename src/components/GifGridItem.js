import React from 'react';

export const GifGridItem = ( {title, url }) => {


  return (
    <div className='overflow-hidden rounded-md border-black border-2 xs:w-1/2 md:w-2/12 animate__animated animate__fadeIn mx-px my-5'>
      <img 
        className='w-full md:h-5/6 sm:h-5/6 xs:h-5/6'
        src={ url }
        alt={ title }
      />
      <p className='my-1 text-center'>{ title }</p>
    </div>
  );
};
