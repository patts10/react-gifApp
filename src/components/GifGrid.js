import React from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifGridItem } from "./GifGridItem";

export const GifGrid = ({ category }) => {

  const { data:images, loading } = useFetchGifs( category );

  

  return (
    <div className="bg-slate-100 p-5">
      <h3 className="text-lg font-bold text-gray-900 sm:text-3xl sm:truncate mt-3 mb-5 mt-5"> { category } </h3>

      { loading && <p className="animate__animated animate__flash">Loading...</p>}

      <div>
        <ol className="flex flex-wrap justify-around">
          {images.map((img) => (
            <GifGridItem key={img.id} {...img} />
          ))}
        </ol>
      </div>
    </div>
  );
};