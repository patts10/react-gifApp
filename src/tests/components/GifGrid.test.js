import { shallow } from 'enzyme'
import { GifGrid } from "../../components/GifGrid"
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');


describe('tests on <GifGrid />', () => { 

  const category = 'pokemon';

  test('should show component correctly', () => { 

    useFetchGifs.mockReturnValue({
      data: [],
      loading: true
    });
    
    const wrapper = shallow(<GifGrid category={ category } />);
    expect(wrapper).toMatchSnapshot();
    
  });
  
  test('should show items when load useFetchGifs images', () => { 

    const gifs = [{
      id:'abc',
      url: 'https//:blab単ab単a.jpg',
      title: 'something'
    },
    {
      id:'123',
      url: 'https//:blab単ab単a.jpg',
      title: 'something'
    }];

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false
    });
    
    const wrapper = shallow(<GifGrid category={ category } />);

    // expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    })


 })