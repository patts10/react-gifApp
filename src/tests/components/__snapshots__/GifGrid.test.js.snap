// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on <GifGrid /> should show component correctly 1`] = `
<div
  className="bg-slate-100 p-5"
>
  <h3
    className="text-lg font-bold text-gray-900 sm:text-3xl sm:truncate mt-3 mb-5 mt-5"
  >
     
    pokemon
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="flex flex-wrap justify-around"
  />
</div>
`;
